packages/flutter/test/material/time_picker_test.dart

```diff
@@ -110,4 +111,92 @@ void main() {
     await finishPicker(tester);
     expect(result.hour, equals(9));
   });
+
+  group('haptic feedback', () {
+    const Duration kFastFeedbackInteral = const Duration(milliseconds: 10);
+    const Duration kSlowFeedbackInteral = const Duration(milliseconds: 200);
+    int hapticFeedbackCount;
+
+    setUpAll(() {
+      PlatformMessages.setMockStringMessageHandler('flutter/platform', (String message) async {
+        if (message == '{"method":"HapticFeedback.vibrate","args":[]}')
```

Comment:
I'd probably make this test slightly looser.  Rather than requiring this exact string, you can use `setMockJSONMessageHandler` and just check that the method on the parsed object is `"HapticFeedback.vibrate"`